lib/assessment.rb

```diff
@@ -5,35 +5,52 @@
 # Create a class called Dog. It should have, as attributes, name, breed, and age.
 # Make all of those attributes read-only.
 
-##
-# your code here
-##
+class Dog
+  attr_reader :name
+  attr_reader :breed
+  attr_reader :age
+  def initialize(name, breed, age)
+    @name = name
+    @breed = breed
+    @age = age
+  end
+end
 
 # Question 2
 # Create a new Dog object.
 
-Answer.dog = nil # replace nil with your code
+Answer.dog = Dog.new( "Max", "Lab", "9")
 
 # Question 3
 # Create a Person class, with attributes name, age, and location.
 # It should set all three attributes when you create a new Person.
 # Name and location should be writeable, but age should be read-only.
 
-##
-# your code here
-##
+class Person
+  attr_accessor :name
+  attr_accessor :location
```

Comment:
Nailed it!!! You're one of few people who got this one perfectly. `attr_accessor` is the right tool here. Although `attr_writer` would give you a 'setter' method, it _won't_ give you a 'setter' method; since it's really unusual to want to be able to write to a variable but not be able to read from it, `attr_writer` is almost never used.